### PR TITLE
Feature/model/delete

### PR DESCRIPTION
**Pull request by iroper-tessella**
Added a feature to delete a model.

Tests and documentation included.

### CODE REVIEW

File: dafni_cli/auth.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            self.set_details_from_dict(auth_dict)
+
+    def set_details_from_dict(self, auth_dict: dict):
+        """Function to set the Auth attributes based on a given auth dictionary
+
+        Args:
+            auth_dict (dict): auth dictionary found in model and dataset dictionaries
+        """
+        self.asset_id = check_key_in_dict(auth_dict, ["asset_id"], default=None)
+        self.destroy = check_key_in_dict(auth_dict, ["destroy"], default=None)
```
Comment (jamie-short) [inline dafni_cli/auth.py:49]: We should default these to False, not None. Same goes for all flags

Comment (iroper-tessella) [reply dafni_cli/auth.py]: Changed

**Comment by jamie-short** [comment]
Can you add an `__init__.py` file into the command folder please

File: test/fixtures/auth_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import pytest
+
+
+@pytest.fixture
+def model_auth_fixture() -> dict:
```
Comment (jamie-short) [inline test/fixtures/auth_fixtures.py:5]: doc string

File: test/fixtures/auth_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      "view": True,
+      "read": True,
+      "update": False,
+      "destroy": False
+    }
+    return auth
+
+
+@pytest.fixture
+def dataset_auth_fixture() -> dict:
```
Comment (jamie-short) [inline test/fixtures/auth_fixtures.py:18]: doc string

File: test/fixtures/auth_fixtures.py (hunk 2 above)
Comment (iroper-tessella) [reply test/fixtures/auth_fixtures.py:5]: Added

File: test/fixtures/auth_fixtures.py (hunk 3 above)
Comment (iroper-tessella) [reply test/fixtures/auth_fixtures.py:24]: Added

**Review by jamie-short — APPROVED** [review]
Changes look good

**Pull request merged**